pennylane/interfaces/execution.py

```diff
@@ -408,6 +393,10 @@ def cost_fn(params, x):
             # disable caching on the forward pass
             execute_fn = qml.interfaces.cache_execute(batch_execute, cache=None)
 
+            # For adjoint backward pass the state needs to be reset
+            if gradient_kwargs["method"] == "adjoint_jacobian":
+                gradient_kwargs["use_device_state"] = False

```

Comment:
It was not working for the backward pass, as the state would already be the final one. I don't think it was a bug but maybe previous backward pass was different. Let me double check that.

Reply:
I found a solution, in the backward mode for gradient device method the jvps were calculated before the results of the function and therefore the state was in the initial state instead of the final state.